Cython/Compiler/PyrexTypes.py

```diff
@@ -1039,6 +1069,34 @@ def specialize(self, values):
     def cast_code(self, expr_code):
         return expr_code
 
+    # incref paths are often special-cased for memoryviews. Therefore pass an explicit
+    # flag to make it clear that memoryviewslices should be included. This provides a path
+    # to making the code more generic, without requiring it to be all done at once
+    def generate_incref(self, cname, do_for_memoryviewslice=False, **kwds):
+        if do_for_memoryviewslice:
+            # FIXME use_utility_code
+            return "__PYX_INC_MEMVIEW(&%s, %d);" % (cname, int(kwds['have_gil']))
+
+    # decref however did look to always apply for memoryview slices
+    # with "have_gil" set to True by default
+    def generate_xdecref(self, cname, have_gil=True, **kwds):
+        # FIXME use_utility_code
+        return "__PYX_XDEC_MEMVIEW(&%s, %d);" % (cname, int(have_gil))
```

Comment:
Swallowing all `**kwargs` always has a code smell because it tends to hide bugs and typos.